### PR TITLE
Replace Lua with LuaJIT and restrict loaded libraries for auto splitters

### PR DESCRIPTION
**Pull request by wins1ey**
Restrict the loaded libraries to only base, string, math and jit. This is based on information from:
http://lua-users.org/wiki/SandBoxes

TODO:
- [x] Further restrict the use of unsafe functions in the base library (e.g. setmetatable)
- [ ] Test current auto splitters with the changes

### CODE REVIEW

**Comment by Loomeh** [comment]
Tony Hawk's Underground seemed to not work. It first made me switch out some logic (I had to change the '&' to 'and' on line 18).

The timer started correctly (loading works too) but LAST didn't split (which is weird considering it printed out the debug message for splitting)
```
Process: THUGONE.exe
PID: 4508
Refresh rate: 120
Started Manhattan; splitting timer...
```

**Comment by wins1ey** [comment]
> The timer started correctly (loading works too) but LAST didn't split (which is weird considering it printed out the debug message for splitting)

Do you get the same issue with JSR? I quickly tested JSR and it was working as expected.
Perhaps this `story_flags[1] = true` is causing a problem and breaking the auto splitter?
What happens if you add another print statement after `return true`

**Comment by EXtremeExploit** [comment]
> Tony Hawk's Underground seemed to not work. It first made me switch out some logic (I had to change the '&' to 'and' on line 18).
The & on that line is a bitwise operation, an "and" isnt the same logic, it will treat them like booleans

**Comment by wins1ey** [comment]
The bit library is now available for auto splitters.
[Lua BitOp](https://bitop.luajit.org/)
[Lua BitOp API functions](https://bitop.luajit.org/api.html)
`local bit = require("bit")` isn't needed because the library is already loaded by LAST.

**Comment by EXtremeExploit** [comment]
Just to have it in hand

Confirmed to work:
- [ ] AmidEvil.lua
- [ ] JSR.lua
- [x] MLP_AMBA.lua
- [ ] mec.lua
- [ ] mecadntp.lua
- [ ] thug.lua

File: src/auto-splitter.c [hunk 1]
```diff
@@ -26,6 +26,24 @@ atomic_bool toggle_loading = false;
 atomic_bool call_reset = false;
 bool prev_is_loading;
 
+static const char* functions[] = {
```
Comment (EXtremeExploit) [inline src/auto-splitter.c:29]: rename this to disabled_functions

**Review by EXtremeExploit — CHANGES_REQUESTED** [review]
just 1 small rename but other than that i see all is good, the change is fast

**Review by EXtremeExploit — APPROVED** [review]

**Comment by Loomeh** [comment]
Mirror's Edge Catalyst's auto splitter doesn't work. LAST complains about most of the readAddress variables returning the wrong return type (expected boolean).

Some logic seems to be working. It does print the debug message for loading, but nothing else works.

**Comment by wins1ey** [comment]
Does #43 fix them?

**Pull request merged**